src/main/java/com/neatresults/mgnltweaks/json/JsonBuilder.java

```diff
@@ -431,6 +438,59 @@ public Object remove(final Object key) {
             return obj;
         }
 
+		@Override
+		public Object get(Object key) {
+			Object superResult = super.get(key);
+			if (!(superResult instanceof ContentMap))
+				return superResult;
+
+			Node node = ((ContentMap) superResult).getJCRNode();
+			if (isArrayParent(node))
+                return childrenAsContentMapList(node);
+
+            return superResult;
+		}
+
+        private boolean isArrayParent(Node candidate) {
+            for (String key : config.childrenArrayCandidates.keySet()) {
+                final String value = config.childrenArrayCandidates.get(key);
+                final Pattern valuePattern = Pattern.compile(value);
+                final Pattern keyPattern = Pattern.compile(key);
+
+                try {
+                    if (!asPropertyStream(candidate.getProperties()).filter(
+							p -> {
+								try {
+									return keyPattern.matcher(p.getName()).matches()
```

Comment:
Yeah, I didn't like it either. The utils method looks like a good solution to me (even though it silently ignores exceptions, but that's the same problem as below).